agent/structs/connect_proxy_config.go

```diff
@@ -37,6 +37,19 @@ type MeshGatewayConfig struct {
 	Mode MeshGatewayMode `json:",omitempty"`
 }
 
+func (c *MeshGatewayConfig) IsZero() bool {
+	zeroVal := MeshGatewayConfig{}
+	return *c == zeroVal
+}
+
+func (base *MeshGatewayConfig) OverlayWith(overlay MeshGatewayConfig) MeshGatewayConfig {
```

Comment:
At first I was curious why this wasn't called `Merge` but after thinking more about it I really like `OverlayWith`.